### PR TITLE
cmd: add test for deocder

### PR DESCRIPTION
**Pull request by imtbkcat**
Add test for decoder function. 
Make decoder more user friendly.

Before:
```
 ./tidb-ctl decoder -f=table_row -k "t\x80\x00\x00\x00\x00\x00\x07\x8f_r\x80\x00\x00\x00\x00\x08\x3b\xba"
table_id: 1935
row_id: 539578
```

After:
```
./tidb-ctl decoder "t\x80\x00\x00\x00\x00\x00\x07\x8f_r\x80\x00\x00\x00\x00\x08\x3b\xba"
format: table_row
table_id: 1935
row_id: 539578
```

Index value is also supported.
```
./tidb-ctl decoder "t\200\000\000\000\000\001p\244_i\200\000\000\000\000\000\000\001\001wms4.0\000\000\375\003\200\000\000\000\006\237@\300"
format: table_index
table_id: 94372
row_id: 1
index_value[0]: {type: bytes, value: wms4.0}
index_value[1]: {type: bigint, value: 111100096}
```

And support TiKV format key at the same time.
```
./tidb-ctl decoder "t\200\000\000\000\000\000\007\217_r\200\000\000\000\000\010;\272"
format: table_row
table_id: 1935
row_id: 539578
```

### CODE REVIEW

**Comment by lysu** [comment]
```
./tidb-ctl decoder "t\200\000\000\000\000\000\025\377\316_r\200\000\001j\331\377\357vI\000\000\000\000\000\372"
Error: illegal base64 data at input byte 1
Usage:
  tidb-ctl decoder [flags]

Flags:
  -h, --help   help for decoder

Global Flags:
      --host ip         TiDB server host (default 127.0.0.1)
      --pdhost ip       PD server host (default 127.0.0.1)
      --pdport uint16   PD server port (default 2379)
      --port uint16     TiDB server port (default 10080)

illegal base64 data at input byte 1
```
it seems still can not decode tikv log output :D

**Comment by lysu** [comment]
it seems be produced by https://github.com/tikv/tikv/blob/8dae71abe25cdcc511116f7d5b29312dffc2b152/components/tikv_util/src/codec/bytes.rs#L22

**Comment by imtbkcat** [comment]
@lysu Why this table id so long, `table_id` is always 8 bytes.

**Comment by imtbkcat** [comment]
@lysu TiKV format table-row key has been supported.

```
./tidb-ctl decoder "t\200\000\000\000\000\000\025\377\316_r\200\000\001j\331\377\357vI\000\000\000\000\000\372"
format: table_row
table_id: 5582
row_id: 1558434510409
```

Could you give some case about `table_index` key in tikv?

**Comment by lysu** [comment]
```
2019/06/17 07:33:32.985 ERRO endpoint.rs:441: Locked(primary_lock: "t\200\000\000\000\000\000\000\255_i\200\000\000\000\000\000\000\001\003\200\000\000\000\000e\221|\003\200\000\000\000\0008\307\024\003\200\000\000\000\0014\025\230\003\200\000\000\000 C\nV" lock_version: 409135484187181177 key: "t\200\000\000\000\000\000\000\255_i\200\000\000\000\000\000\000\002\003\200\000\000\000\0014\025\230\003\200\000\000\000\000e\221|\003\200\000\000\000\0008\307\024" lock_ttl: 3031)
```


@imtbkcat  it seems still some kv log key cannot be decode :rofl:

**Comment by lysu** [comment]
```
./tidb-ctl decoder "t\200\000\000\000\000\000\000\255_i\200\000\000\000\000\000\000\002\003\200\000\000\000\0014\025\230\003\200\000\000\000\002eWm\003\200\000\000\000\000\024\301\210"
format: table_index
table_id: 173
row_id: 2
index_value[0]: {type: bigint, value: 20190616}
index_value[1]: {type: bigint, value: 40195949}
index_value[2]: {type: bigint, value: 1360264}
```

**Comment by lysu** [comment]
LGTM

File: cmd/decoder_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package cmd
```
Comment (jackysp) [inline cmd/decoder_test.go:14]: please add the license

**Review by jackysp — COMMENTED** [review]
rest LGTM

**Review by jackysp — APPROVED** [review]
LGTM

**Pull request merged**